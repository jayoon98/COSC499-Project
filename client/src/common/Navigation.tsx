import React, { useEffect, useState } from 'react';
import {
  View,
  StyleSheet,
  TouchableOpacity,
  Dimensions,
  Platform,
} from 'react-native';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { Actions } from 'react-native-router-flux';
import { IconProp } from '@fortawesome/fontawesome-svg-core';

import {
  faCalendar,
  faNewspaper,
  faCircle,
  faUser,
} from '@fortawesome/free-regular-svg-icons';

import {
  faCalendar as faCalendarSolid,
  faNewspaper as faNewspaperSolid,
  faCircle as faCircleSolid,
  faUser as faUserSolid,
} from '@fortawesome/free-solid-svg-icons';
import { ThemeContext, themes } from './ThemeContext';

const screen = Dimensions.get('screen');

export type NavigationProps = {
  children?: React.ReactNode;
  selected?: string; // This is no longer required
};

//export const navHeight = screen.height * 0.065;

export let navHeight = 80;

if (Platform.OS !== 'ios') {
  navHeight = screen.height * 0.065;
}

type NavButtonProps = {
  name: string;
  style: any;
};

// TODO: Find better icons for these
const icons = {
  circles: [faCircle, faCircleSolid],
  questionnaires: [faNewspaper, faNewspaperSolid],
  calendar: [faCalendar, faCalendarSolid],
  settings: [faUser, faUserSolid],
};

// Adds a navigation bar to a component. Very bare bones right now.
// example:
//     <Navigation selected="pageName">
//       <MyComponent />
//     </Navigation>
export function Navigation({ children, selected }: NavigationProps) {
  const [currentScene, setCurrentScene] = useState(Actions.currentScene);

  // Scene loading is async so we can't just use currentScene directly, instead
  // we watch for when the scene changes to set the currently selected icon
  // in the nav bar.
  useEffect(() => {
    setCurrentScene(Actions.currentScene);
  }, [setCurrentScene, Actions.currentScene]);

  if (!selected) {
    selected = currentScene;
  }

  const NavButton = ({ name, style }: NavButtonProps) => {
    const icon = icons[name][name === selected ? 1 : 0];
    return (
      <TouchableOpacity
        style={{ ...styles.navIcon, ...style }}
        onPress={() => Actions.replace(name)}
      >
        <FontAwesomeIcon icon={icon} size={28} color="#12121a" />
      </TouchableOpacity>
    );
  };

  return (
    <View style={styles.container}>
      <View style={{ marginBottom: navHeight }}>{children}</View>
      <View style={styles.navigation}>
        <NavButton name="circles" style={{ flexGrow: 1, paddingLeft: 32 }} />
        <NavButton
          name="questionnaires"
          style={{ flexGrow: 2, alignItems: 'center' }}
        />
        <NavButton
          name="calendar"
          style={{ flexGrow: 2, alignItems: 'center' }}
        />
        <NavButton
          name="settings"
          style={{ flexGrow: 1, alignItems: 'flex-end', paddingRight: 32 }}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  navigation: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 'auto',
    backgroundColor: 'white',
    height: navHeight,
    padding: 8,
    paddingLeft: 0,
    paddingRight: 0,
    borderTopWidth: 1,
    borderColor: '#eaeaea',
  },
  navIcon: {
    flexGrow: 1,
    display: 'flex',
  },
});
