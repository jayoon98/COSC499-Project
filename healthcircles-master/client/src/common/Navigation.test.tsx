import React from 'react';
import renderer from 'react-test-renderer';
import { Navigation } from './Navigation';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

it('renders correctly', () => {
  const tree = renderer.create(<Navigation />).toJSON();
  expect(tree).toMatchSnapshot();
});
