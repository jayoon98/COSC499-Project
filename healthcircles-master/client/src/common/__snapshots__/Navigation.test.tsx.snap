// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 80,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#eaeaea",
        "borderTopWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "space-between",
        "marginTop": "auto",
        "padding": 8,
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "display": "flex",
          "flexGrow": 1,
          "opacity": 1,
          "paddingLeft": 32,
        }
      }
    >
      <
        color="#12121a"
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f111",
              "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z",
            ],
            "iconName": "circle",
            "prefix": "far",
          }
        }
        size={28}
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexGrow": 2,
          "opacity": 1,
        }
      }
    >
      <
        color="#12121a"
        icon={
          Object {
            "icon": Array [
              576,
              512,
              Array [],
              "f1ea",
              "M552 64H112c-20.858 0-38.643 13.377-45.248 32H24c-13.255 0-24 10.745-24 24v272c0 30.928 25.072 56 56 56h496c13.255 0 24-10.745 24-24V88c0-13.255-10.745-24-24-24zM48 392V144h16v248c0 4.411-3.589 8-8 8s-8-3.589-8-8zm480 8H111.422c.374-2.614.578-5.283.578-8V112h416v288zM172 280h136c6.627 0 12-5.373 12-12v-96c0-6.627-5.373-12-12-12H172c-6.627 0-12 5.373-12 12v96c0 6.627 5.373 12 12 12zm28-80h80v40h-80v-40zm-40 140v-24c0-6.627 5.373-12 12-12h136c6.627 0 12 5.373 12 12v24c0 6.627-5.373 12-12 12H172c-6.627 0-12-5.373-12-12zm192 0v-24c0-6.627 5.373-12 12-12h104c6.627 0 12 5.373 12 12v24c0 6.627-5.373 12-12 12H364c-6.627 0-12-5.373-12-12zm0-144v-24c0-6.627 5.373-12 12-12h104c6.627 0 12 5.373 12 12v24c0 6.627-5.373 12-12 12H364c-6.627 0-12-5.373-12-12zm0 72v-24c0-6.627 5.373-12 12-12h104c6.627 0 12 5.373 12 12v24c0 6.627-5.373 12-12 12H364c-6.627 0-12-5.373-12-12z",
            ],
            "iconName": "newspaper",
            "prefix": "far",
          }
        }
        size={28}
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexGrow": 2,
          "opacity": 1,
        }
      }
    >
      <
        color="#12121a"
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f133",
              "M400 64h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V112c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V160h352v298c0 3.3-2.7 6-6 6z",
            ],
            "iconName": "calendar",
            "prefix": "far",
          }
        }
        size={28}
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "display": "flex",
          "flexGrow": 1,
          "opacity": 1,
          "paddingRight": 32,
        }
      }
    >
      <
        color="#12121a"
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f007",
              "M313.6 304c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 304 0 364.2 0 438.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-25.6c0-74.2-60.2-134.4-134.4-134.4zM400 464H48v-25.6c0-47.6 38.8-86.4 86.4-86.4 14.6 0 38.3 16 89.6 16 51.7 0 74.9-16 89.6-16 47.6 0 86.4 38.8 86.4 86.4V464zM224 288c79.5 0 144-64.5 144-144S303.5 0 224 0 80 64.5 80 144s64.5 144 144 144zm0-240c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z",
            ],
            "iconName": "user",
            "prefix": "far",
          }
        }
        size={28}
      />
    </View>
  </View>
</View>
`;
